packages/analytics/src/utils.ts

```diff
@@ -40,3 +40,21 @@ export const encodeDataToQueryString = <T extends Event>(
 
     return params.toString();
 };
+
+export const reportEventWithRetry = async (
```

Comment:
nit: I would prefer passing more params to function as an object to make the parameters "named" (`retry: false` instead of just `false`)
```javascript
interface ReportEventProps {
    type: Event['type'];
    url: string;
    options: RequestInit;
    retry: boolean;
}

export const reportEvent = async ({ type, url, options, retry = true }: ReportEventProps) => {
    try {
        await fetch(url, options);
    } catch (err) {
        if (!retry) {
            console.error(`Analytics report failed. Reporting '${type}' was unsuccessful. ${err}`);
            return;
        }
        console.error(`Analytics report failed. Reporting '${type}' again. ${err}`);
        reportEvent({ type, url, options, retry: false });
    }
};
```

Reply:
something like this? 8e4efbaad33269109823384ac3e782cc2fefca11 together with error change